### PR TITLE
DiagnosticUrlBuilder now makes URLs that go to the readme versus wiki

### PR DESCRIPTION
**Pull request by twofingerrightclick**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [![NuGet Status](http://img.shields.io/nuget/v/IntelliTect.Analyzers.svg?style=flat&label=IntelliTect.Analyzers)](https://www.nuget.org/packages/IntelliTect.Analyzers/)
 
-[![Build Status](https://intellitect.visualstudio.com/CodingStandards/_apis/build/status/IntelliTect.CodingStandards?branchName=master)](https://intellitect.visualstudio.com/CodingStandards/_build/latest?definitionId=76&branchName=master)
+[![Build Status](https://intellitect.visualstudio.com/CodingGuidelines/_apis/build/status/IntelliTect.CodingGuidelines?branchName=master)](https://intellitect.visualstudio.com/CodingGuidelines/_build/latest?definitionId=76&branchName=master)
```
Comment (twofingerrightclick) [inline README.md:3]:
```suggestion
[![Build Status](https://intellitect.visualstudio.com/CodingStandards/_apis/build/status/IntelliTect.CodingStandards?branchName=master)](https://intellitect.visualstudio.com/CodingStandards/_build/latest?definitionId=76&branchName=master)
```

Comment (twofingerrightclick) [reply README.md:3]: The pipeline title will probably need to be updated.

File: IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
     {
-        private const string BaseUrl = "https://github.com/IntelliTect/CodingStandards/wiki/";
+        private const string BaseUrl = "https://github.com/IntelliTect/CodingGuidelines";
 
         /// <summary>
         /// Get the full diagnostic help url
         /// </summary>
-        /// <param name="analyzerBlock">Current analyzer block</param>
-        /// <param name="intlCode">The intl error code</param>
+        /// <param name="title">intl analyzer title</param>
```
Comment (Keboo) [inline IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs:15]: For comments we should avoid abbreviations.
```suggestion
        /// <param name="title">IntelliTect analyzer title</param>
```

File: IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+        /// <param name="diagnosticId">The intl error code</param>
```
Comment (Keboo) [inline IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs:16]:
```suggestion
        /// <param name="diagnosticId">The IntelliTect error code</param>
```

File: IntelliTect.Analyzer/IntelliTect.Analyzer/IntelliTect.Analyzer.csproj [hunk 4]
```diff
@@ -8,9 +8,9 @@
   <PropertyGroup>
     <PackageId>IntelliTect.Analyzers</PackageId>
     <Authors>IntelliTect</Authors>
-    <RepositoryUrl>https://github.com/IntelliTect/CodingStandards</RepositoryUrl>
+    <RepositoryUrl>https://github.com/IntelliTect/CodingGuidelines</RepositoryUrl>
     <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
-    <Description>Analyzer enforcing IntelliTect internal coding standards</Description>
+    <Description>Analyzer aiding the implementation of IntelliTect internal coding guidelines</Description>
```
Comment (Keboo) [inline IntelliTect.Analyzer/IntelliTect.Analyzer/IntelliTect.Analyzer.csproj:13]: 👍

File: README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 [![Build Status](https://intellitect.visualstudio.com/CodingStandards/_apis/build/status/IntelliTect.CodingStandards?branchName=master)](https://intellitect.visualstudio.com/CodingStandards/_build/latest?definitionId=76&branchName=master)
 
-# CodingStandards / Design Standards
+# CodingGuidelines / Design Guidelines
```
Comment (Keboo) [inline README.md:5]:
```suggestion
# Coding Guidelines / Design Guidelines
```

File: IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
-            return BaseUrl + analyzerBlock.GetDescription() + $"#{intlCode}";
+            if (string.IsNullOrWhiteSpace(title))
+                throw new System.ArgumentException("title cannot be empty", nameof(title));
+            if (string.IsNullOrWhiteSpace(diagnosticId))
+                throw new System.ArgumentException("diagnostic ID cannot be empty", nameof(diagnosticId));
+
+            Regex hyphenateRegex = new Regex(@"\s");
+            string hyphenatedTitle = hyphenateRegex.Replace(title, "-");
+
+            return BaseUrl + $"#{diagnosticId.ToLower()}" + $"---{hyphenatedTitle.ToLower()}";
```
Comment (Keboo) [inline IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs:29]:
```suggestion
            return BaseUrl + $"#{diagnosticId.ToLowerInvariant()}" + $"---{hyphenatedTitle.ToLowerInvariant()}";
```

Comment (twofingerrightclick) [reply IntelliTect.Analyzer/IntelliTect.Analyzer/DiagnosticUrlBuilder.cs:29]: Thanks. I looked up the invariant principle

**Pull request merged**